src/compiler/crystal/config.cr

```diff
@@ -15,12 +15,17 @@ module Crystal
     end
 
     def self.description
-      formatted_sha = "[#{build_commit}] " if build_commit
+      details = [version]
+      details << "[#{build_commit}]" if build_commit
+      details << "(#{date})" unless date.empty?
```

Comment:
Using an array + join is fine, we can merge it with that. But it's suboptimal.
A better way would be what we had before, just with leading whitespace added to each optional piece. Or `String.build` could also be an alternative.
This is an optional suggestion, my approval doesn't depend on it.

Reply:
I updated this with String.build just now